source/distributed/shared_tria.cc

```diff
@@ -283,13 +283,12 @@ namespace parallel
       {
         // Assert that each cell is owned by a processor
         unsigned int n_my_cells = 0;
-        typename parallel::shared::Triangulation<dim,
-                                                 spacedim>::active_cell_iterator
-          cell = this->begin_active(),
-          endc = this->end();
-        for (; cell != endc; ++cell)
-          if (cell->is_locally_owned())
-            n_my_cells += 1;
+        for (const auto &cell : this->active_cell_iterators() |
+                                  IteratorFilters::LocallyOwnedCell())
+          {
+            (void)cell;
+            ++n_my_cells;
```

Comment:
It is disappointing that one has to give the loop variable a name, but that appears to be the case based on the standard.

Reply:
Would it be more elegant to use `std::count_if` here?

Reply:
Or use what you did above in `source/distributed/repartitioning_policy_tools.cc`.
```
const auto locally_owned_cells =
    tria_in.active_cell_iterators() | IteratorFilters::LocallyOwnedCell();
const unsigned int n_locally_owned_active_cells =
    std::count(locally_owned_cells.begin(), locally_owned_cells.end());
```